apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java

```diff
@@ -118,6 +118,11 @@
          * Limit the length of the operationName to prevent errors when inserting elasticsearch
          **/
         public static int OPERATION_NAME_THRESHOLD = 500;
+
+        /**
+         * If true, skywalking agent will enable profile when user create a new profile task
+         */
+        public static boolean ACTIVE_PROFILE = false;
```

Comment:
As a key feature of 7.0.0, this flag should be `true` as default. @arugal Could you share your concern?

Reply:
> As a key feature of 7.0.0, this flag should be `true` as default. @arugal Could you share your concern?

1.  I think thread monitoring is an auxiliary function
1.  Thread snapshot collection has a great impact on application performance

So I think it should be enabled by the user display

Reply:
That is why we set the acceptable task with the following limitations
1. Only one endpoint at a certain time.
1. 15 mins as max duration of the task
1. dumpPeriod >= 10ms
1. minDurationThreshold for collecting thread dump in meaningful case.
1. maxSamplingCount should less than 10. FYI @mrproliu I think `maxSamplingCount` is missing from existing codes, backend and protocol. right? Please fix this.

We should not set that in the false, because once the undetected performance issue happens, you have no change to reboot agent already.

Reply:
You could recheck these in the design doc, except the (5), we miss that in the doc too.

Reply:
> You could recheck these in the design doc, except the (5), we miss that in the doc too.

I'll re-read the design document

Reply:
@mrproliu Set up a config class for profile first.

```java
public static class Profile {
           public static boolean ACTIVE = true;
}
```

Reply:
please check. `maxSamplingCount` have added into design document and codes.